hassio/src/ingress-view/hassio-ingress-view.ts

```diff
@@ -151,41 +156,71 @@ class HassioIngressView extends LitElement {
       addon = await fetchHassioAddonInfo(this.hass, addonSlug);
     } catch (err: any) {
       await showAlertDialog(this, {
-        text: "Unable to fetch add-on info to start Ingress",
+        text: this.supervisor.localize("ingress.error_addon_info"),
         title: "Supervisor",
       });
       await nextRender();
-      history.back();
+      navigate("/hassio/store", { replace: true });
       return;
     }
 
-    if (!addon.ingress_url) {
+    if (!addon.version) {
       await showAlertDialog(this, {
-        text: "Add-on does not support Ingress",
+        text: this.supervisor.localize("ingress.error_addon_not_installed"),
         title: addon.name,
       });
       await nextRender();
-      history.back();
+      navigate(`/hassio/addon/${addon.slug}/info`, { replace: true });
       return;
     }
 
-    if (addon.state !== "started") {
+    if (!addon.ingress_url) {
       await showAlertDialog(this, {
-        text: "Add-on is not running. Please start it first",
+        text: this.supervisor.localize("ingress.error_addon_not_supported"),
         title: addon.name,
       });
       await nextRender();
-      navigate(`/hassio/addon/${addon.slug}/info`, { replace: true });
+      history.back();
       return;
     }
 
+    if (addon.state !== "started") {
+      const confirm = await showConfirmationDialog(this, {
+        text: this.supervisor.localize("ingress.error_addon_not_running"),
+        title: addon.name,
+        confirmText: this.supervisor.localize("ingress.start_addon"),
+        dismissText: this.supervisor.localize("common.no"),
+      });
+      if (confirm) {
+        try {
+          await startHassioAddon(this.hass, addonSlug);
+          // Give the add-on some time to start
+          await new Promise((resolve) => {
+            setTimeout(resolve, 1000);
+          });
```

Comment:
This will never be a stable value.
1: Query the SU API until the add-on has "started"
2: Query the ingress path, untill it resolves to something other than 502

This can have a timeout if desired, but with this approach, maybe something more than 1s, 30s maybe?
In most cases it will be faster

Reply:
Is there a way to get the healthy state of an addon?

Reply:
Could that be added?

Reply:
Probably, but it would not help much. As each addon would need to be updated to declare that. Which will not happen.

Reply:
I understood from Frenck that a lot of them already have it?

Reply:
So, the main issue right now (for the frontend purpose of things), is that an add-on can be either "started", "stopped" or "error".

For these purposes, it would be better if there was a "starting" and "stopping" state in general as well. The Supervisor could leverage the healthy container state to help with determining the difference between "starting" and "started" for that as well, during the container startup (awaiting the first healthy report, like Docker itself does as well).

I agree, that not all add-ons will/do/are able to provide that. In general, it might be best if it was exposed if an add-on has health checks or not, so we can act accordingly.

Reply:
That would indeed be better, but we still would need something like <https://github.com/home-assistant/frontend/pull/16458#discussion_r1187140857> for those who do not have that.

Reply:
I don't think it does, that's why I added the retry dialog on the `load` event of the iframe, it think it is fine to fallback to that if health checks are not supported?

Reply:
> 1: Query the SU API until the add-on has "started"

This is not needed, as we await the start command, it will await the actual start of the addon.

Reply:
> I agree, that not all add-ons will/do/are able to provide that. In general, it might be best if it was exposed if an add-on has health checks or not, so we can act accordingly.

If this helps it's possible to do this. This info can be obtained by inspecting the container.

Without a healthcheck though I'm not entirely sure how we could do this. I poked around in `s6-rc` a bit to see if there's a way we can obtain the current state of `s6` services to put a healthcheck in the base container but doesn't look like there's much info available on services in the container (and not every addon uses s6 anyway).

A timeout sounds good but the value will vary wildly. Many addons are quick but some with a lot of data take a while to actually get started.

Reply:
For my idea, how many addons have health checks? And is it something that is easily added?

Reply:
For most add-ons adding health checks is a matter of adding a single configuration statement to the Dockerfile.

Reply:
Upstream has been merged: https://github.com/home-assistant/supervisor/pull/4376

Add-ons will now have a state: `startup` while still starting, and `started` when that is finished.

This will work best for add-ons that implement a health check of course.

../Frenck